src/main/java/org/jenkinsci/plugins/rundeck/RundeckNotifier.java

```diff
@@ -131,30 +139,36 @@ public Object readResolve() {
     }
 
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
-            throws InterruptedException, IOException {
-        if (build.getResult() != Result.SUCCESS) {
-            return true;
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
+        if (run.getResult() != Result.SUCCESS && run.getResult() != null) {
+            return;
         }
 
         RundeckClient rundeck = getDescriptor().getRundeckInstance(this.rundeckInstance);
 
         if (rundeck == null) {
             listener.getLogger().println("Rundeck configuration is not valid !");
-            return false;
+            throw new AbortException("Rundeck configuration is not valid !");
         }
         try {
             rundeck.ping();
         } catch (RundeckApiException e) {
             listener.getLogger().println("Rundeck is not running !");
-            return false;
+            throw new AbortException("Rundeck is not running !");
         }
 
-        if (shouldNotifyRundeck(build, listener)) {
-            return notifyRundeck(rundeck, build, listener);
+        if (shouldNotifyRundeck(run, listener)) {
+            notifyRundeck(rundeck, run, listener);
         }
+    }
 
-        return true;
+    private ChangeLogSet<? extends Entry> getChangeSet(@Nonnull Run<?, ?> run) {
+        if(run instanceof AbstractBuild<?,?>) {
```

Comment:
You can amend your commit locally and push it with `--force` to do not create an additional commit in your PR